pkg/client/record/event.go

```diff
@@ -42,49 +55,82 @@ type EventRecorder interface {
 // your binary. Subsequent calls will be ignored. The return value can be ignored
 // or used to stop recording, if desired.
 func StartRecording(recorder EventRecorder, source api.EventSource) watch.Interface {
+	// Set up our own personal buffer of events so that we can clear out GetEvents'
+	// broadcast channel as quickly as possible to avoid causing the relatively more
```

Comment:
That sounds fine-- perhaps you can add a ActionIfNotFull method to broadcaster. Thanks!